### PR TITLE
Jacoco coverrage Threshold rouding error

### LINKED ISSUE #10752: Jacoco coverrage Threshold rouding error
When really close to 100% coverage you can have rounding errors causing build not to fail with missed instructions. Known bug on Jacoco's side, can be "fixed" by changing: 

```
<limit>
  <counter>LINE</counter>
  <value>COVEREDRATIO</value>
  <minimum>1.00</minimum>
  <value>MISSEDCOUNT</value>
  <maximum>0</maximum>
</limit>
```

to 

```
<limit>
  <counter>BRANCH</counter>
  <value>COVEREDRATIO</value>
  <minimum>1.00</minimum>
  <value>MISSEDCOUNT</value>
  <maximum>0</maximum>
</limit>
```

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #10752

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1292,7 +1292,7 @@
                       <element>CLASS</element>
                       <limits>
                         <limit>
-                          <counter>LINE</counter>
+                          <counter>BRANCH</counter>
```
Comment (murdos) [inline pom.xml:1295]: Then it becomes a duplicated limit of the limited defined below

Comment (qmonmert) [reply pom.xml:1295]: so what's your suggestion?

Comment (murdos) [reply pom.xml:1295]: I don't know, I've not done any investigation to fix the mentioned issue.
I'm just noticing that this limit is already defined.
Maybe @DamnClin know better.

Comment (qmonmert) [reply pom.xml:1295]: ok

Comment (DamnClin) [reply pom.xml:1295]: The idea is to go from 

```
<value>COVEREDRATIO</value>
<minimum>1.00</minimum>
```

To 

```
<value>MISSEDCOUNT</value>
<maximum>0</maximum>
```

For both banch and line, I missed copy past in the ticket, forgot to remove the failing part...

Comment (murdos) [reply pom.xml:1295]: @qmonmert : That means `COVEREDRATIO`  limits should be removed

File: src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java [hunk 2]
```diff
@@ -199,14 +199,14 @@ private static GradleMainBuildPlugin gradleJacocoWithMinCoverageCheckPlugin() {
 
                   limit {
                       counter = "LINE"
-                      value = "COVEREDRATIO"
-                      minimum = "1.00".toBigDecimal()
+                      value = "MISSEDCOUNT"
+                      maximum = "1.00".toBigDecimal()
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java:203]: should be 0 no?

File: src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java [hunk 3]
```diff
@@ -199,14 +199,14 @@ private static GradleMainBuildPlugin gradleJacocoWithMinCoverageCheckPlugin() {
...
                   }
 
                   limit {
                       counter = "BRANCH"
-                      value = "COVEREDRATIO"
-                      minimum = "1.00".toBigDecimal()
+                      value = "MISSEDCOUNT"
+                      maximum = "1.00".toBigDecimal()
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java:209]: should be 0 no?

File: src/test/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactoryTest.java [hunk 4]
```diff
@@ -228,14 +228,14 @@ void shouldBuildJacocoWithMinCoverageCheckModule() {
 
                   limit {
                       counter = "LINE"
-                      value = "COVEREDRATIO"
-                      minimum = "1.00".toBigDecimal()
+                      value = "MISSEDCOUNT"
+                      maximum = "1.00".toBigDecimal()
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactoryTest.java:232]: should be 0 no?

File: src/test/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactoryTest.java [hunk 5]
```diff
@@ -228,14 +228,14 @@ void shouldBuildJacocoWithMinCoverageCheckModule() {
...
                   }
 
                   limit {
                       counter = "BRANCH"
-                      value = "COVEREDRATIO"
-                      minimum = "1.00".toBigDecimal()
+                      value = "MISSEDCOUNT"
+                      maximum = "1.00".toBigDecimal()
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactoryTest.java:238]: should be 0 no?

**Review by DamnClin — APPROVED** [review]

**Pull request merged**